torch/nn/utils/rnn.py

```diff
@@ -331,7 +331,7 @@ def pad_packed_sequence(
     unsorted_indices = sequence.unsorted_indices
     if unsorted_indices is not None:
         batch_dim = 0 if batch_first else 1
-        return padded_output.index_select(batch_dim, unsorted_indices), lengths[unsorted_indices]
+        return padded_output.index_select(batch_dim, unsorted_indices), lengths[unsorted_indices.cpu()]
```

Comment:
Wow! This is surprising. Shouldn't unsorted_indices only be transferred to CPU if lengths is on CPU?